### PR TITLE
Allows Predators to make and wear bones

### PR DESCRIPTION
**Pull request by BeagleGaming1**
# About the pull request

This lets predators take limbs, use their knife on it, then turn those into bones that they can attach to their armor.
Also moves a bit of other pred stuff to its place (why was the globe with the turfs?)

Sprites by sleepynecrons
# Explain why it's good for the game

This gives predators another use for limbs they get.

and someone asked me to make it
# Changelog

:cl:
add: Allows Yautja to make bones from limbs and wear them
/:cl:

### CODE REVIEW

File: code/modules/cm_preds/yaut_items.dm [hunk 1]
```diff
@@ -923,3 +923,79 @@
 		icon_state = "medicomp_open"
 	else
 		icon_state = "medicomp"
+
+/obj/item/polishing_rag
+	name = "polishing rag"
+	desc = "An astonishingly fine, hand-tailored piece of exotic cloth."
```
Comment (carlarctg) [inline code/modules/cm_preds/yaut_items.dm:929]: Give this a hint that it's used for bones, maybe as a pred only text in get_examine_tect

File: code/modules/cm_preds/yaut_items.dm [hunk 2]
```diff
@@ -923,3 +923,79 @@
...
+	name = "arm bone"
+	icon_state = "r_arm"
+/obj/item/clothing/accessory/limb/skeleton/r_foot
+	name = "foot bone"
+	icon_state = "r_foot"
+/obj/item/clothing/accessory/limb/skeleton/r_hand
+	name = "hand bone"
+	icon_state = "r_hand"
+/obj/item/clothing/accessory/limb/skeleton/r_leg
+	name = "leg bones"
```
Comment (carlarctg) [inline code/modules/cm_preds/yaut_items.dm:980]: either name them all bone or all bones

File: code/modules/cm_preds/yaut_items.dm [hunk 3]
```diff
@@ -923,3 +923,79 @@
...
+	icon_state = "r_leg"
+/obj/item/clothing/accessory/limb/skeleton/head
+	name = "skull"
+	icon_state = "head"
+	high_visibility = TRUE
+/obj/item/clothing/accessory/limb/skeleton/torso
+	name = "rib"
```
Comment (carlarctg) [inline code/modules/cm_preds/yaut_items.dm:987]: ribcage

also the skull should still have the spine like in avp

File: code/modules/cm_preds/yaut_weapons.dm [hunk 4]
```diff
@@ -492,7 +492,26 @@
...
 		hide.amount = skin_amount
 
+/obj/item/weapon/melee/yautja/knife/afterattack(obj/potential_limb, mob/living/user, proximity)
+	. = ..()
+	if(!proximity)
+		return
+	if(!HAS_TRAIT(user, TRAIT_YAUTJA_TECH))
+		return ..()
+
+	if(istype(potential_limb, /obj/item/limb)) //Not checking for torso and groin, if you somehow get those, good on you
```
Comment (harryob) [inline code/modules/cm_preds/yaut_weapons.dm:502]: early return if it's not of type

File: code/modules/cm_preds/yaut_items.dm [hunk 5]
```diff
@@ -923,3 +923,90 @@
...
+	. = ..()
+	if(HAS_TRAIT(user, TRAIT_YAUTJA_TECH))
+		. += "You could use this to polish bones."
+
+/obj/item/polishing_rag/afterattack(obj/potential_limb, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!HAS_TRAIT(user, TRAIT_YAUTJA_TECH))
+		return
+
+	if(istype(potential_limb, /obj/item/clothing/accessory/limb/skeleton)) //Not checking for torso and groin, if you somehow get those, good on you
```
Comment (harryob) [inline code/modules/cm_preds/yaut_items.dm:943]: early return

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/cm_preds/yaut_machines.dm [hunk 6]
```diff
@@ -1,7 +1,57 @@
...
+	icon = 'icons/obj/structures/machinery/yautja_machines.dmi'
+	stored_material =  list("metal" = 40000, "glass" = 20000)
+
+/obj/structure/machinery/prop/yautja/bubbler
+	name = "yautja cauldron"
+	desc = "A large, black machine emitting an ominous hum with an attached pot of boiling fluid. Bits of what appears to be leftover lard and balls of hair can be seen floating inside of it."
+	icon = 'icons/obj/structures/machinery/yautja_machines.dmi'
+	///What bones will come from the vat
+	var/next_limb
+	var/base_state = "vat"
```
Comment (harryob) [inline code/modules/cm_preds/yaut_machines.dm:21]: does this ever change? if not use initial()

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]
sprites

**Review by nauticall — CHANGES_REQUESTED** [review]
sprites are all fine 'cept for the towel, that looks kinda blocky. would recommend using something like this if you would like, or you can make your own variation of it, up to you.......

that's the only thing this PR needs before the ole' rubber stamp (on the sprite side at least)

image

**Comment by nauticall** [comment]
there is that pred gear dmi conflict though, might wanna look at that

**Review by harryob — APPROVED** [review]

**Pull request merged**